epm/Distribution/Client/IndexUtils.hs

```diff
@@ -450,7 +451,7 @@ packageIndexFromCache mkPkg hnd entrs mode = accum mempty [] entrs
       -- from the index tarball if it turns out that we need it.
       -- Most of the time we only need the package id.
       ~(pkg, pkgtxt) <- unsafeInterleaveIO $ do
-        mPatch <- patchedPackageCabalFile pkgid
+        mPatch <- patchedPackageCabalFile pkgid defaultPatchesDir
```

Comment:
So this is tricky because that it will mess up future calls to `git pull`. We need symlink behavior here and we need a way to revert to the default. One way is protect `git pull` is - if we find that there's something bad, bail out, delete the entire `patches` directory, and reclone the repository. So if you run `epm update` after `epm update --patches-directory=[something]`, your `patches` directory will be fixed. This seems a bit hacky - is it worth it?